### DISCUSSION TITLE
Cyrilic nicknames in irc side

### DISCUSSION THREAD

**Discussion by BCC-BGChatCommunity** [qaisjp/go-discord-irc#105] [Q&A] [answered]
I want to make some changes from irc side. Our server have cyrilic nicknames like абвгд I found the letter script in irc/nick/charAttrs and include the whole alphabet but the script accept this type on discord side only. My question is what i need to change to make this happend ? 
Thanks

**Comment by llmII** [comment, marked as answer]
@h1x0r1 - You're saying if a nick 'абвгд' exists on IRC server and sends a message "message" that it doesn't get relayed into discord?

If so (trying to check that I understood the issue properly), the other part you describe sounds like it's only used by the bridge to be sure to generate puppets that acquire a nick that is compatible with _any_ IRC server. 

If it's not sending IRC -> Discord when nick is like 'абвгд' is it printing any errors? If not then I guess you'd need to look into a) debugging (I'm unsure how to do this with go, so can't help) or b) print debugging (instrumentation, and I can explain this one a bit). If you go with option b you need to find where '*han <-' is in the code and surround it by fmt.Printf/fmt.Println code that captures the data around the call and prints it out so you can see what happens. If you find the printing functions don't get encountered then need to search up the call stack from where '*han <-' is and insert prints until you find where the creation of the message that happens before it's sent is dropped and thus never sent.

I don't have a way to test against cyrllic nicks or I'd try to be more helpful :(

**Reply by BCC-BGChatCommunity** [reply]
Thanks for the answer. My problem is different. When i make nickchange in diskord from "sdfff" to "абв" bot auto make nickchange in irc side with "abv" He translate nick in english alphabet not get the actual cyrilic nickname that i make.  How to edit this to tell him to not translate those nicknames in english.

**Reply by llmII** [reply]
Odd question, are you certain the IRCd supports nicks in crylic? (Might want to check ISUPPORT or others to determine).

If the answer is yes and if it's in ISUPPORT if you could strip identifying info like "Network" out of ISUPPORT and then paste it here I might would be interested.

Basically you almost need a setting that says "don't change characters in the nick if they match these characters". It's something to be careful about because can't just allow any character (be very interesting to see someone insert a NUL byte). You probably don't want emoji's in IRC Nicks (not sure if any IRCd supports that and I sincerely hope none ever will). If ISUPPORT reports this ordeal then might could look at building a table centered around ISUPPORT that adds in characters for whatever setting is in there... but if not probably are wanting to modify the (I think searching Snowflake will help, but this is from memory, so could be wrong) code to add a setting to support specifying a string full of "allowed nick characters" and having that override the default builtin one when specified.

**Reply by qaisjp** [reply]
I think the relevant code is here: https://github.com/qaisjp/go-discord-irc/blob/33a4c606f1f30311344db594b08fda406cbac407/bridge/irc_manager.go#L271-L274
